python/pyspark/pandas/frame.py

```diff
@@ -8689,7 +8702,13 @@ def sample(
         sdf = self._internal.resolved_copy.spark_frame.sample(
             withReplacement=replace, fraction=frac, seed=random_state
         )
-        return DataFrame(self._internal.with_new_sdf(sdf))
+        if ignore_index:
+            spark_index_names = [
+                SPARK_INDEX_NAME_FORMAT(i) for i in range(self._internal.index_level)
+            ]
+            return DataFrame(sdf.drop(*spark_index_names))
```

Comment:
```suggestion
            return DataFrame(sdf.drop(*self._internal.index_spark_column_names))
```

I think maybe `spark_index_names` is already in `self._internal.index_spark_column_names` ?